config/dhcp_relay.py

```diff
@@ -0,0 +1,158 @@
+import click
+import ipaddress
+import utilities_common.cli as clicommon
+import utilities_common.dhcp_relay_util as dhcp_relay_util
+
+
+DHCP_RELAY_TABLE = "DHCP_RELAY"
+DHCPV6_SERVERS = "dhcpv6_servers"
+IPV6 = 6
+
+VLAN_TABLE = "VLAN"
+DHCPV4_SERVERS = "dhcp_servers"
+IPV4 = 4
+
+
+def validate_ips(ctx, ips, ip_version):
+    for ip in ips:
+        try:
+            ip_address = ipaddress.ip_address(ip)
+        except Exception:
+            ctx.fail("{} is invalid IP address".format(ip))
+
+        if ip_address.version != ip_version:
+            ctx.fail("{} is not IPv{} address".format(ip, ip_version))
+
+
+def get_dhcp_servers(db, vlan_name, ctx, table_name, dhcp_servers_str):
+    table = db.cfgdb.get_entry(table_name, vlan_name)
+    if len(table.keys()) == 0:
+        ctx.fail("{} doesn't exist".format(vlan_name))
+
+    dhcp_servers = table.get(dhcp_servers_str, [])
+
+    return dhcp_servers, table
+
+
+def add_dhcp_relay(vid, dhcp_relay_ips, db, ip_version):
+    table_name = DHCP_RELAY_TABLE if ip_version == 6 else VLAN_TABLE
+    dhcp_servers_str = DHCPV6_SERVERS if ip_version == 6 else DHCPV4_SERVERS
+    vlan_name = "Vlan{}".format(vid)
+    ctx = click.get_current_context()
+    # Verify ip addresses are valid
+    validate_ips(ctx, dhcp_relay_ips, ip_version)
+    dhcp_servers, table = get_dhcp_servers(db, vlan_name, ctx, table_name, dhcp_servers_str)
+    added_ips = []
+
+    for dhcp_relay_ip in dhcp_relay_ips:
+        # Verify ip addresses not duplicate in add list
+        if dhcp_relay_ip in added_ips:
+            ctx.fail("Error: Find duplicate DHCP relay ip {} in add list".format(dhcp_relay_ip))
+        # Verify ip addresses not exist in DB
+        if dhcp_relay_ip in dhcp_servers:
+            click.echo("{} is already a DHCP relay for {}".format(dhcp_relay_ip, vlan_name))
+            return
+
+        dhcp_servers.append(dhcp_relay_ip)
+        added_ips.append(dhcp_relay_ip)
+
+    table[dhcp_servers_str] = dhcp_servers
+
+    db.cfgdb.set_entry(table_name, vlan_name, table)
+    click.echo("Added DHCP relay address [{}] to {}".format(",".join(dhcp_relay_ips), vlan_name))
+    try:
+        dhcp_relay_util.restart_dhcp_relay_service()
```

Comment:
extract to `def handle_restart_dhcp_relay_service()`?

Reply:
Fixed